backend/accounts/tests.py

```diff
@@ -0,0 +1,336 @@
+
+        # Assert access and refresh tokens exist
+        self.assertIn('access', response.data)
+        self.assertIn('refresh', response.data)
+
+        # Add additional assertions to check the response data
+        account_detail = response.data['user']
+        self.assertEqual(account_detail['id'], 1)
+        self.assertEqual(account_detail['email'], 'chris@email.com')
+        self.assertEqual(account_detail['first_name'], "")
+        self.assertEqual(account_detail['last_name'], "")
+        self.assertEqual(account_detail['gender'], "")
+        self.assertIsNone(account_detail['dob'])
+
+        # Check if user exists in database
+        self.assertTrue(get_user_model().objects.filter(email="chris@email.com", is_active=True).exists())
+
+    def test_registration_with_invalid_email(self):
+
+        registration_data = {
+            'email': 'chris',
+            'password1': 'testpass123',
+            'password2': 'testpass123',
+        }
+
+        response = self.client.post('/api/v1/accounts/registration/', registration_data)
+
+        self.assertEqual(
+            str(response.data["email"][0]),
+            "Enter a valid email address."
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_registration_with_empty_password(self):
+
+        registration_data = {
+            'email': 'chris@email.com',
+            'password1': '',
+            'password2': '',
+        }
+
+        response = self.client.post('/api/v1/accounts/registration/', registration_data)
+
+        self.assertEqual(str(response.data["password1"][0]), "This field may not be blank.")
+        self.assertEqual(str(response.data["password2"][0]), "This field may not be blank.")
+
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_registration_with_invalid_password(self):
+
+        registration_data = {
+            'email': 'chris@email.com',
+            'password1': 'short',
+            'password2': 'short',
+        }
+
+        response = self.client.post('/api/v1/accounts/registration/', registration_data)
+
+        self.assertIn("This password is too short.", str(response.data["password1"][0]))
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+
+class GoogleSigninTest(TestCase):
+    """Test suite for signing in with Google"""
+
+    def setUp(self):
+        self.client = APIClient()
+
+        # Google provides some default values through the social account adaptor
+        self.EXPECTED_RESPONSE = {
+            "status_code": status.HTTP_201_CREATED,
+            "data": {
+                "access": "ACCESS_TOKEN",
+                "refresh": "REFRESH_TOKEN",
+                "user": {
+                    "id": 1,
+                    "email": "christopherfkk@uni.minerva.edu",
+                    "username": "kar_keung_christopher_fok",  # Google fill
```

Comment:
so google also provides this right? what if @kar_keung_christopher_fok already exists in our database?